main.go

```diff
@@ -7,134 +7,98 @@ import (
 	"net/http"
 	"os"
 	"os/signal"
-	"syscall"
 	"time"
 
 	"golang.org/x/sync/errgroup"
 
-	"github.com/cristalhq/aconfig"
+	"github.com/alexflint/go-arg"
 	"github.com/go-logr/logr"
 	"github.com/go-logr/zapr"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/spf13/afero"
+	"github.com/xenitab/pkg/kubernetes"
 	"go.uber.org/zap"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
 
 	"github.com/xenitab/git-auth-proxy/pkg/auth"
 	"github.com/xenitab/git-auth-proxy/pkg/config"
 	"github.com/xenitab/git-auth-proxy/pkg/server"
 	"github.com/xenitab/git-auth-proxy/pkg/token"
 )
 
-type cfg struct {
-	Addr           string `flag:"addr" default:":8080" required:"true"`
-	MetricsAddr    string `flag:"metrics-addr" default:":9090" required:"true"`
-	ConfigPath     string `flag:"config" required:"true"`
-	KubeconfigPath string `flag:"kubeconfig"`
-}
-
 func main() {
+	var args struct {
+		Addr           string `arg:"--addr,required" default:":8080"`
+		MetricsAddr    string `arg:"--metrics-addr,required" default:":9090"`
+		CfgPath        string `arg:"--config,required"`
+		KubeconfigPath string `arg:"--kubeconfig"`
+	}
+	arg.MustParse(&args)
```

Comment:
Should we use something that panics?

I usually separate it out to a newConfig() like this:
```go
func newConfig(args []string) (Config, error) {
	cfg := Config{}

	parser, err := arg.NewParser(arg.Config{
		Program:   "azcagit",
		IgnoreEnv: false,
	}, &cfg)
	if err != nil {
		return Config{}, err
	}

	err = parser.Parse(args)
	if err != nil {
		return Config{}, err
	}

	return cfg, nil
}
```

And then in main:
```go
	cfg, err := config.NewConfig(os.Args[1:])
	if err != nil {
		log.Error(err, "unable to load config")
		os.Exit(1)
	}
```

Reply:
I personally think this is fine beacuse the function does a bunch of other usefult stuff like printing helper info to stdout.

Reply:
I will have another look and see if there is a better way of doing this.

Reply:
@simongottschlag isn't logging an error and then calling `os.Exit()` basically functionally equivalent to panicking though? The end result of both designs is that your application dies on startup when config is missing.

Reply:
https://go-proverbs.github.io/

> Don't panic

Reply:
I checked the code now and dont really see where the code would panic? It does an `os.Exit(1)`